src/bank01_player.asm

```diff
@@ -697,18 +693,57 @@ scrollRoom:
     set  2, A                                          ;; 01:4502 $cb $d7
     ld   [wMainGameStateFlags.nextFrame], A            ;; 01:4504 $ea $a2 $c0
     ld   [wMainGameStateFlags], A                      ;; 01:4507 $ea $a1 $c0
-.jr_01_450a:
-    ld   A, E                                          ;; 01:450a $7b
-    push DE                                            ;; 01:450b $d5
-    bit  0, A                                          ;; 01:450c $cb $47
-    jr   NZ, .west                                     ;; 01:450e $20 $31
-    bit  1, A                                          ;; 01:4510 $cb $4f
-    jp   NZ, .east                                     ;; 01:4512 $c2 $97 $45
-    bit  2, A                                          ;; 01:4515 $cb $57
-    jp   NZ, .south                                    ;; 01:4517 $c2 $e7 $45
-    bit  3, A                                          ;; 01:451a $cb $5f
-    jp   NZ, .north                                    ;; 01:451c $c2 $3a $46
-    pop  DE                                            ;; 01:451f $d1
+    ld   A, D
+    ld   [wSpriteScrollSpeed], A
+
+    ; Set B to be the 'flip' of the direction in E
+    ld   B, $01
+    bit  1, E ; east
+    jr   NZ, .check_room_override
+    sla  B
+    bit  0, E ; west
+    jr   NZ, .check_room_override
+    sla  B
+    bit  3, E ; north
+    jr   NZ, .check_room_override
+    sla  B
+.check_room_override:
+    ld   HL, wNextRoomOverride
+    push DE
+    ld   A, [HL+]
+    ld   E, A
+    ld   D, [HL]
+    and  A, D
+    inc  A
+    ld   A, B ; B=$01 (E), $02 (W), $04 (N), or $08 (S)
+    jr   Z, .choose_next_room
+    or   A, $80
+.choose_next_room:
+    or   A, $10
+    call call_00_2617
+    pop  DE
+    ld   A, $ff
+    ld   [wNextRoomOverride], A
+    ld   [wNextRoomOverride.x], A
+    ld   C, A ; indicates first pass through scrollRoom function
+.scroll_started:
+    ld   A, E
+    and  A, $0f
+    jr   Z, .stop_scroll
+    ld   A, [wScrollPixelCounter]
+    and  A, $0f
+    or   A, C ; on first pass we know we will not be moving the screen
+    jr   NZ, scrollRoomMove
+    ld   A, [wTileCopyRequestCount]
+    and  A, A
+    ret  NZ
+    ld   A, [wBackgroundRenderRequestCount]
+    and  A, A
+    ret  NZ
+    jr   scrollRoomMove
+.stop_scroll:
+    ; not sure if this branch is ever really called, may be able to remove
```

Comment:
My old test run using Beaten Dying Moon shows that during my playthrough I never exercised this path:
```
01:4464 .code:00bb
01:4541 .code:006f
```
That isn't to say there is no path or that it should be eliminated yet, but there is no evidence that it _isn't_ dead code.

Reply:
Hmm, I can't think of a case when it would be used. I'll do some tracing to see if I can prove it's impossible. If so, do you see any reason to keep it? Only thing that comes to mind is if you wanted to script a half-scroll in a cinematic scene, followed by a loadMap.

Reply:
I looked at all the callers - they all ensure direction is set. There isn't a way for this to be called without a direction set. I removed the extraneous block. Saved 54 more bytes!